### PR TITLE
Support for non-public clouds for EventHub scaler.

### LINKED ISSUE #1915: Support non-public clouds for Azure Event Hub
### Proposal

Support non-public clouds for Azure Event Hub

### Use-Case

_No response_

### Anything else?

Relates to #1285

### PR DESCRIPTION
**Pull request by v-shenoy**
Added support for three new fields in the trigger metadata for the eventhub scaler.  The `cloud` parameter can be `azureChinaCloud`, `azureGermanCloud`, `azureUSGovernmentCloud`, `azurePublicCloud`, or `private` (default is `azurePublicCloud`).

```yaml
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: eventhub
  labels: {}
spec:
  scaleTargetRef:
    name: eventhub
  triggers:
  - type: azure-eventhub
    metadata:
      direction: in
      eventHubNamespace: non-public-clouds-ns
      eventHubName: python-hub
      consumerGroup: $Default
      storageConnectionFromEnv: AzureWebJobsStorage
      cloud: "private"
      endpointSuffix: <service-bus-suffix>
      activeDirectoryEndpoint: <ad-endpoint>
    authenticationRef:
      name: eventhub-trigger-auth
---
```

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #1915, #2781

### CODE REVIEW

File: pkg/scalers/azure_eventhub_scaler.go [hunk 1]
```diff
@@ -121,6 +122,29 @@ func parseAzureEventHubMetadata(config *ScalerConfig) (*eventHubMetadata, error)
 		meta.eventHubInfo.BlobContainer = val
 	}
 
+	meta.eventHubInfo.Cloud = azure.DefaultCloud
+	if val, ok := config.TriggerMetadata["cloud"]; ok {
+		meta.eventHubInfo.Cloud = val
+	}
+
+	serviceBusEndpointSuffixProvider := func(env az.Environment) (string, error) {
+		return env.ServiceBusEndpointSuffix, nil
```
Comment (JorTurFer) [inline pkg/scalers/azure_eventhub_scaler.go:131]: is ServiceBus correct here?

**Review by JorTurFer — COMMENTED** [review]
LGTM,
only small nits inline

File: pkg/scalers/azure_eventhub_scaler.go (hunk 1 above)
Comment (v-shenoy) [reply pkg/scalers/azure_eventhub_scaler.go:131]: If you look at the event hub namespace information in the public cloud, the host name for an Event Hub namespace is <name-of-namespace>.servicebus.windows.net. You can see that the event hub Go SDK uses the service bus endpoint suffix, [here](https://github.com/Azure/azure-event-hubs-go/blob/73b7c0f7b25625f14c9a877ecb73230eba886107/namespace.go#L69).

Comment (v-shenoy) [reply pkg/scalers/azure_eventhub_scaler.go:131]: The `namespaceWithAzureEnvironment` is used to create the namespace in the `NewHub` function which we use for getting the event hub client.

Comment (JorTurFer) [reply pkg/scalers/azure_eventhub_scaler.go:131]: thanks for explaining this, I though that it was an error ❤️

**Review by JorTurFer — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]
LGTM

**Pull request merged**